### PR TITLE
pr for Playlist

### PR DESCRIPTION
**Pull request by suryxks**
features implemented:

- add to playlist
- delete from playlist
- create playlist
- delete playlist 
- add to play list from video page

please refresh the page once after login if you are testing the features implemented

### CODE REVIEW

File: src/AppRoutes.js [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
       <Route path='/:videoId' element={<VideoPlayerPage />} />
       <Route path='/liked' element={<Liked />} />
       <Route path='/history' element={<History />} />
+      <Route path='playList/:playlistId' element={<PlaylistPage />} />
       <Route path='/playList' element={<PlayList />} />
       <Route path='/watchlater' element={<WatchLater />} />
       <Route path='/signup' element={<Signup />} />
       <Route path='/login' element={<Login />} />
+      <Route path='/' element={<Home />} />
       <Route path='*' element={
```
Comment (poornima-alugubelly) [inline src/AppRoutes.js:19]: Nice that u added match not found case as well

File: src/components/CardMenu/CardMenu.jsx [hunk 2]
```diff
@@ -6,11 +6,14 @@ import { useLocation } from 'react-router-dom';
 import { useOnClickOutside } from '../../hooks/useOnClickOutside';
 import './CardMenu.css'
 import { useData } from '../../contexts/DataContext';
-export const CardMenu = ({ isMenuOpen, setIsMenuOpen, id, video }) => {
+import {PlaylistModal} from '../PlayListModal/PlayListModal'
+import { usePlayList } from '../../contexts/PlayListContext';
+export const CardMenu = ({ isMenuOpen, setIsMenuOpen, id, video}) => {
     const ref = useRef();
```
Comment (poornima-alugubelly) [inline src/components/CardMenu/CardMenu.jsx:12]: Nice! Didn't know that modal is a usecase for useref learnt something new!

File: src/components/PlaylistCard/PlaylistCard.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import DeleteIcon from '@mui/icons-material/Delete';
+import PlaylistAddIcon from '@mui/icons-material/PlaylistAdd';
+import {Link} from 'react-router-dom'
+import { usePlayList } from '../../contexts/PlayListContext';
+import './PlaylistCard.css'
+export const PlaylistCard = ({ img, name, numberOfvideos ,id}) => {
+    const {deletePlayList}=usePlayList();
+    return (
```
Comment (poornima-alugubelly) [inline src/components/PlaylistCard/PlaylistCard.jsx:8]: Cool use of custom hooks for updating playlist

**Review by poornima-alugubelly — APPROVED** [review]

**Review by someshtiwarii — APPROVED** [review]
clean code, looks good to go

**Pull request merged**